prow/bugzilla/client.go

```diff
@@ -214,37 +231,112 @@ func getRootForClone(c Client, bug *Bug) (*Bug, error) {
 
 // GetRootForClone returns the original bug.
 func (c *client) GetRootForClone(bug *Bug) (*Bug, error) {
-	return getRootForClone(c, bug)
+	bugCache := map[int]*Bug{}
+	return getRootForClone(c, bug, bugCache)
 }
 
 // GetAllClones returns all the clones of the bug including itself
 // Differs from GetClones as GetClones only gets the child clones which are one level lower
-func (c *client) GetAllClones(bug *Bug) ([]*Bug, error) {
-	return getAllClones(c, bug)
+func (c *client) GetAllClones(bugID int) ([]*Bug, *Bug, *Bug, error) {
+	bugCache := map[int]*Bug{}
+	return getAllClones(c, bugID, bugCache)
 }
-func getAllClones(c Client, bug *Bug) ([]*Bug, error) {
-	root, err := getRootForClone(c, bug)
+
+type clonesCacheWithMux struct {
+	cache map[int][]int
+	mux   sync.Mutex
+}
+
+var clonesCache = clonesCacheWithMux{cache: map[int][]int{}}
```

Comment:
This must be threadsafe, you can not use package-level variables for this. Also, lets pelase do this proper and do for all caches we have a:
```
type thingCache struct {
  cache map[string]thing
  lock sync.Mutext
}

func (tc *thingCache)Get(key string) (thing, bool) {
  tc.lock.Lock()
  defer tc.lock.Unlock()
  return tc.cache[key]
}

func (tc *thingCache) set(key string, item thing) {
  tc.lock.Lock()
  defer tc.lock.Unlock()
  tc.cache[key] = item
}
```